python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py

```diff
@@ -26,14 +26,20 @@
         is_required=False,
         default_value=5,
     ),
+    "profile_name": Field(
+        str,
+        description="Specifies a profile to connect that session",
+        is_required=False,
+        default_value="default",
```

Comment:
This default value is what's causing most of these failing tests. AWS has a number of [different ways to provide credentials](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#configuring-credentials). By always passing a profile, we're forcing the credentials lookup to short circuit at step 4 and expect that there's a profile defined as `default`.

These buildkite hosts don't have a `~/.aws/credentials` file; consequently it can't find a `default` profile defined in that file. Hence all the `botocore.exceptions.ProfileNotFound: The config profile (default) could not be found` errors.

I think we should instead treat this more like we treat `endpoint_url` or `region_name` - completely optional and no default (which will cause it to default to `None`). That way, when we call `boto3.session.Session(profile_name=profile_name)`, it'll succeed regardless of whether a profile name is passed or not.

I suspect once you make that update, you'll see the overwhelming majority of these tests start passing.

Reply:
Thank you for suggestions! I will update the code to reflect that.

Reply:
Done!